### ISSUE TITLE
Orientdb 3.0 - Problem with restarts when db is closed abruptly

### ISSUE THREAD

**Issue by careerscale** [orientechnologies/orientdb#7489]
## Expected behavior  
Database should stabilize and recover from any failures that might have caused because of abrupt close of db   

## Actual behavior  
I think I have no idea why some times if db is closed (command window is closed) without control C, it doesn't work.

This never comes out of it. If it i doing some processing like fixing indexes, it must show status and let user be aware of it. going non responsive doesn't make sense at all.

```
2017-06-12 09:02:41:257 WARNI Record OFuzzyCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=16167388}} com.orientechnologies.orient.core.stora
ge.impl.local.paginated.wal.OFuzzyCheckpointStartRecord{lsn=null, previousCheckpoint=OLogSequenceNumber{segment=0, position=16167341}} will be skipped during 
data restore [OLocalPaginatedStorage]                                                                                                                         
2017-06-12 09:02:41:257 WARNI Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointEndRecord{lsn=OLogSequenceNumber{segme
nt=0, position=16167428}} will be skipped during data restore [OLocalPaginatedStorage]                                                                        
2017-06-12 09:02:41:258 WARNI Record OFuzzyCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=16167435}} com.orientechnologies.orient.core.stora
ge.impl.local.paginated.wal.OFuzzyCheckpointStartRecord{lsn=null, previousCheckpoint=OLogSequenceNumber{segment=0, position=16167388}} will be skipped during 
data restore [OLocalPaginatedStorage]                                                                                                                         
2017-06-12 09:02:41:259 WARNI Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointEndRecord{lsn=OLogSequenceNumber{segme
nt=0, position=16167475}} will be skipped during data restore [OLocalPaginatedStorage]                                                                        
2017-06-12 09:02:41:260 WARNI Record OFuzzyCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=16167482}} com.orientechnologies.orient.core.stora
ge.impl.local.paginated.wal.OFuzzyCheckpointStartRecord{lsn=null, previousCheckpoint=OLogSequenceNumber{segment=0, position=16167435}} will be skipped during 
data restore [OLocalPaginatedStorage]                                                                                                                         
2017-06-12 09:02:41:261 WARNI Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointEndRecord{lsn=OLogSequenceNumber{segme
nt=0, position=16167522}} will be skipped during data restore [OLocalPaginatedStorage]                                                                        
2017-06-12 09:02:41:261 WARNI Record OFuzzyCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=16167529}} com.orientechnologies.orient.core.stora
ge.impl.local.paginated.wal.OFuzzyCheckpointStartRecord{lsn=null, previousCheckpoint=OLogSequenceNumber{segment=0, position=16167482}} will be skipped during 
data restore [OLocalPaginatedStorage]                                                                                                                         
2017-06-12 09:02:41:262 WARNI Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointEndRecord{lsn=OLogSequenceNumber{segme
nt=0, position=16167569}} will be skipped during data restore [OLocalPaginatedStorage]                                                                        
2017-06-12 09:02:41:263 INFO  Storage data recover was completed [OLocalPaginatedStorage]                                                                     
2017-06-12 09:02:44:302 INFO  {db=mlm} Wait till indexes restore after crash was finished. [OIndexManagerShared]       


```                       
## Steps to reproduce  
Try having active databases created and used and then try killing db few times while active transactions are going on and try restart. you may be able to reproduce.

**Comment by careerscale** [comment]
@luigidellaquila  FYI

**Comment by laa** [comment]
It should be fixed once we merge micro transactions which should happen during this week

**Comment by careerscale** [comment]
@laa 
Thanks for the update. eagerly waiting for the merge into 3.0

**Comment by vsspt** [comment]
I verified the same problem, was actually going to open an issue for this.

**Comment by kindofwhat** [comment]
if it helps: run the following (kotlin) script and kill the process anytime
`

import org.apache.tinkerpop.gremlin.orientdb.OrientGraphFactory

fun main(args: Array<String>) {

    val o = OrientGraphFactory("plocal:C:/windows/temp/testdb").tx

    for (i in 0..1000) {
        o.addVertex("label")
        println("now would be a good time to kill the process")
        Thread.sleep(5000)
    }
}
`

**Comment by careerscale** [comment]
restarting the terminal and running server.bat again helped us in windows.

**Comment by laa** [comment]
it should be already fixed in 3.0 version after we merged micro-txs and fixed restore procedure.
Could you check the latest snapshot?

**Comment by careerscale** [comment]
@laa 

"restarting the terminal and running server.bat again helped us in windows."  this still holds good for me.

but when I hit the issue crash with indexes, I still see the same problem

**Comment by laa** [comment]
@careerscale what version do you use?  In the latest snapshot indexes are not rebuilt on startup. Do you use the version built from sources?

**Comment by careerscale** [comment]
it is the latest snapshot.

**Comment by vsspt** [comment]
I've used the 3.0.0m1 version.

**Comment by laa** [comment]
@careerscale could you provide an output of the log which you get after the crash and restore, do you use Lucene indexes? Could you also send us thread dump at the moment when index rebuild is frozen? Could you provide thread dump as separate gist?

**Comment by careerscale** [comment]
I am not using anything specific with Lucene.

Here is the log. latest with July 4th build

16:36:47.739 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Storage 'mlm' was not closed properly. Will try to recover from write ahead log
16:36:47.739 [http-nio-8084-exec-1] INFO  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Looking for last checkpoint...
16:36:47.740 [http-nio-8084-exec-1] INFO  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Found FUZZY checkpoint.
16:36:47.740 [http-nio-8084-exec-1] INFO  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Data restore procedure from FUZZY checkpoint is started.
16:36:47.741 [http-nio-8084-exec-1] INFO  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - 1 operations were processed, current LSN is OLogSequenceNumber{segment=0, position=16} last LSN is OLogSequenceNumber{segment=0, position=21139129}
16:36:47.743 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Non tx operation was used during data modification we will need index rebuild.
16:36:47.747 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=5272}, previousCheckpoint=null} will be skipped during data restore
16:36:47.748 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=5280}} will be skipped during data restore
16:36:47.753 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=13994}, previousCheckpoint=OLogSequenceNumber{segment=0, position=5272}} will be skipped during data restore
16:36:47.754 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=14018}} will be skipped during data restore
16:36:47.768 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=38896}, previousCheckpoint=OLogSequenceNumber{segment=0, position=13994}} will be skipped during data restore
16:36:47.768 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=38920}} will be skipped during data restore
16:36:47.849 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=131828}, previousCheckpoint=OLogSequenceNumber{segment=0, position=38896}} will be skipped during data restore
16:36:47.850 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=131852}} will be skipped during data restore
16:36:47.889 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=727281}, previousCheckpoint=OLogSequenceNumber{segment=0, position=131828}} will be skipped during data restore
16:36:47.890 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=727305}} will be skipped during data restore
16:36:47.907 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=1119276}, previousCheckpoint=OLogSequenceNumber{segment=0, position=727281}} will be skipped during data restore
16:36:47.908 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=1119300}} will be skipped during data restore
16:36:47.917 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=1339359}, previousCheckpoint=OLogSequenceNumber{segment=0, position=1119276}} will be skipped during data restore
16:36:47.917 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=1339383}} will be skipped during data restore
16:36:47.924 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=1592983}, previousCheckpoint=OLogSequenceNumber{segment=0, position=1339359}} will be skipped during data restore
16:36:47.925 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=1593007}} will be skipped during data restore
16:36:47.940 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=2107049}, previousCheckpoint=OLogSequenceNumber{segment=0, position=1592983}} will be skipped during data restore
16:36:47.940 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=2107073}} will be skipped during data restore
16:36:47.949 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=2432109}, previousCheckpoint=OLogSequenceNumber{segment=0, position=2107049}} will be skipped during data restore
16:36:47.949 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=2432133}} will be skipped during data restore
16:36:47.956 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=2621497}, previousCheckpoint=OLogSequenceNumber{segment=0, position=2432109}} will be skipped during data restore
16:36:47.957 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=2621521}} will be skipped during data restore
16:36:47.964 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=2864243}, previousCheckpoint=OLogSequenceNumber{segment=0, position=2621497}} will be skipped during data restore
16:36:47.964 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=2864267}} will be skipped during data restore
16:36:47.972 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=3068979}, previousCheckpoint=OLogSequenceNumber{segment=0, position=2864243}} will be skipped during data restore
16:36:47.972 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=3069003}} will be skipped during data restore
16:36:48.021 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=4954666}, previousCheckpoint=OLogSequenceNumber{segment=0, position=3068979}} will be skipped during data restore
16:36:48.022 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=4954690}} will be skipped during data restore
16:36:48.031 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=5231024}, previousCheckpoint=OLogSequenceNumber{segment=0, position=4954666}} will be skipped during data restore
16:36:48.031 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=5231048}} will be skipped during data restore
16:36:48.067 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFullCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=7282100}, previousCheckpoint=OLogSequenceNumber{segment=0, position=5231024}} will be skipped during data restore
16:36:48.067 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=7282124}} will be skipped during data restore
16:36:48.450 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record OFuzzyCheckpointStartRecord{lsn=OLogSequenceNumber{segment=0, position=21139083}} com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointStartRecord{lsn=null, previousCheckpoint=OLogSequenceNumber{segment=0, position=7282100}} will be skipped during data restore
16:36:48.450 [http-nio-8084-exec-1] WARN  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Record com.orientechnologies.orient.core.storage.impl.local.paginated.wal.OFuzzyCheckpointEndRecord{lsn=OLogSequenceNumber{segment=0, position=21139123}} will be skipped during data restore
16:36:48.450 [http-nio-8084-exec-1] INFO  com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage - Storage data recover was completed
16:36:51.856 [http-nio-8084-exec-1] INFO  com.orientechnologies.orient.core.index.OIndexManagerShared - $ANSI{green {db=mlm}} Wait till indexes restore after crash was finished.

**Comment by careerscale** [comment]
@laa ,  threaddump here, I think it should be straight forward to reproduce, I just stop my application and start again.
https://gist.github.com/careerscale/be46b284519ad1a0da0d51900ff08b48

**Comment by careerscale** [comment]
@laa, do you need anything else here?

**Comment by laa** [comment]
Hi @careerscale , no. Right now the problem is clear. It is not related to storage restore process itself but is related to new database lifecycle 

in this thread 
```
	at com.orientechnologies.orient.core.index.OIndexManagerShared.waitTillIndexRestore(OIndexManagerShared.java:324)
	at com.orientechnologies.orient.core.index.OIndexManagerShared.recreateIndexes(OIndexManagerShared.java:303)
	at com.orientechnologies.orient.core.index.OIndexManagerProxy.recreateIndexes(OIndexManagerProxy.java:200)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.initAtFirstOpen(ODatabaseDocumentEmbedded.java:352)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.internalOpen(ODatabaseDocumentEmbedded.java:125)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.internalOpen(ODatabaseDocumentEmbedded.java:110)
	at com.orientechnologies.orient.core.db.OrientDBEmbedded.open(OrientDBEmbedded.java:129)
```
lock on OrientDBEmbedded instance is taken

and in this thread 
```
"OrientDB rebuild indexes" #57 daemon prio=5 os_prio=0 tid=0x0000000028900800 nid=0x6214 waiting for monitor entry [0x000000002d6de000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at com.orientechnologies.orient.core.db.OrientDBEmbedded.openNoAuthenticate(OrientDBEmbedded.java:104)
	- waiting to lock <0x00000006c594b548> (a com.orientechnologies.orient.core.db.OrientDBEmbedded)
	at com.orientechnologies.orient.server.OServer.openDatabase(OServer.java:841)
	at com.orientechnologies.orient.server.OSystemDatabase.openSystemDatabase(OSystemDatabase.java:90)
```
we wait for the same lock. But first thread is waits for second thread. So we have got dealock.
It is not related to storage restore procedure but to the new lifecycle so assigned to @tglman .

**Comment by careerscale** [comment]
Thanks @laa for the analysis. @tglman  when you resolve the issue, please do let me know so that I can validate it.

**Comment by tglman** [comment]
hi @careerscale,

Just pushed a fix that should solve the problem.

Regards

**Comment by careerscale** [comment]
@tglman , Please check this one

Could you please confirm if this has your fix?


2017-07-06 10:57:53
Full thread dump Java HotSpot(TM) 64-Bit Server VM (25.121-b13 mixed mode):

"http-nio-8084-exec-2" #58 daemon prio=5 os_prio=0 tid=0x000000001cf9b000 nid=0x2360 waiting for monitor entry [0x000000002edda000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at com.orientechnologies.orient.core.db.OrientDBEmbedded.exists(OrientDBEmbedded.java:267)
	- waiting to lock <0x00000006c595e4f8> (a com.orientechnologies.orient.core.db.OrientDBEmbedded)
	at com.orientechnologies.orient.core.db.OrientDB.createIfNotExists(OrientDB.java:255)
	at com.orientechnologies.orient.core.db.OrientDB.createIfNotExists(OrientDB.java:242)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraphFactory.getDatabase(OrientGraphFactory.java:151)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraph.<init>(OrientGraph.java:145)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraph.<init>(OrientGraph.java:140)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraphFactory.getGraph(OrientGraphFactory.java:96)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraphFactory.getTx(OrientGraphFactory.java:85)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraphFactory.getTx(OrientGraphFactory.java:89)
	at com.callidus.mlm.config.MLMOrientDbConfig.getOrientGraph(MLMOrientDbConfig.java:90)
	at com.callidus.mlm.config.MLMOrientDbConfig$$EnhancerBySpringCGLIB$$8776ee0e.CGLIB$getOrientGraph$1(<generated>)
	at com.callidus.mlm.config.MLMOrientDbConfig$$EnhancerBySpringCGLIB$$8776ee0e$$FastClassBySpringCGLIB$$4bb2be3b.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:358)
	at com.callidus.mlm.config.MLMOrientDbConfig$$EnhancerBySpringCGLIB$$8776ee0e.getOrientGraph(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1173)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1067)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:325)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.obtainBeanInstanceFromFactory(ConfigurationClassEnhancer.java:389)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:361)
	at com.callidus.mlm.config.MLMOrientDbConfig$$EnhancerBySpringCGLIB$$8776ee0e.getOrientGraph(<generated>)
	at com.callidus.mlm.data.dao.impl.BaseDaoImpl.getOrientGraph(BaseDaoImpl.java:44)
	at com.callidus.mlm.service.internal.CompanyServiceImpl.createUpdate(CompanyServiceImpl.java:218)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:333)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:85)
	at com.callidus.mlm.audit.ApplicationAuditor.serviceMethodExecutionTime(ApplicationAuditor.java:42)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:629)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:618)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:213)
	at com.sun.proxy.$Proxy87.createUpdate(Unknown Source)
	at com.callidus.mlm.resource.CompanyResource.createUpdate(CompanyResource.java:143)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:133)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:116)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:230)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:474)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:349)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:783)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:798)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1434)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	- locked <0x0000000771a01810> (a org.apache.tomcat.util.net.NioEndpoint$NioSocketWrapper)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- <0x000000076cde9500> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"OrientDB rebuild indexes" #57 daemon prio=5 os_prio=0 tid=0x000000001cf9a000 nid=0x1924 waiting for monitor entry [0x000000002ef3e000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at com.orientechnologies.orient.core.db.OrientDBEmbedded.openNoAuthenticate(OrientDBEmbedded.java:102)
	- waiting to lock <0x00000006c595e4f8> (a com.orientechnologies.orient.core.db.OrientDBEmbedded)
	at com.orientechnologies.orient.server.OServer.openDatabase(OServer.java:841)
	at com.orientechnologies.orient.server.OSystemDatabase.openSystemDatabase(OSystemDatabase.java:90)
	at com.orientechnologies.orient.server.OSystemDatabase.execute(OSystemDatabase.java:97)
	at com.orientechnologies.orient.server.security.ODefaultServerSecurity.getSystemUser(ODefaultServerSecurity.java:286)
	at com.orientechnologies.orient.core.metadata.security.OSecurityExternal.getUser(OSecurityExternal.java:92)
	at com.orientechnologies.orient.core.metadata.security.OSecurityProxy.getUser(OSecurityProxy.java:115)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.internalOpen(ODatabaseDocumentEmbedded.java:135)
	at com.orientechnologies.orient.core.index.OIndexManagerShared$RecreateIndexesTask.run(OIndexManagerShared.java:488)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"OrientDB Write Cache Flush Task (mlm)" #54 daemon prio=10 os_prio=2 tid=0x000000001cf99800 nid=0xacc runnable [0x000000002fa8f000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x000000078d18f138> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"OrientDB WAL Flush Task (mlm)" #53 daemon prio=5 os_prio=0 tid=0x000000001cf98800 nid=0x38c0 runnable [0x000000002e57f000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x000000078d12da70> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"WAL Closer Task (mlm)" #52 daemon prio=5 os_prio=0 tid=0x000000001cf98000 nid=0x1734 waiting on condition [0x000000002f8df000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x000000078d12d220> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"Thread-16" #51 daemon prio=5 os_prio=0 tid=0x000000001cf8d800 nid=0x39cc waiting on condition [0x000000002f0be000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x000000078d28d8a8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1067)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"http-nio-8084-exec-1" #50 daemon prio=5 os_prio=0 tid=0x000000001cf8e000 nid=0x1fb4 in Object.wait() [0x000000002146a000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000077119e008> (a java.lang.Thread)
	at java.lang.Thread.join(Thread.java:1249)
	- locked <0x000000077119e008> (a java.lang.Thread)
	at java.lang.Thread.join(Thread.java:1323)
	at com.orientechnologies.orient.core.index.OIndexManagerShared.waitTillIndexRestore(OIndexManagerShared.java:324)
	at com.orientechnologies.orient.core.index.OIndexManagerShared.recreateIndexes(OIndexManagerShared.java:303)
	at com.orientechnologies.orient.core.index.OIndexManagerProxy.recreateIndexes(OIndexManagerProxy.java:201)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.initAtFirstOpen(ODatabaseDocumentEmbedded.java:352)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.internalOpen(ODatabaseDocumentEmbedded.java:125)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.internalOpen(ODatabaseDocumentEmbedded.java:110)
	at com.orientechnologies.orient.core.db.OrientDBEmbedded.open(OrientDBEmbedded.java:127)
	- locked <0x00000006c595e4f8> (a com.orientechnologies.orient.core.db.OrientDBEmbedded)
	at com.orientechnologies.orient.core.db.OrientDBEmbedded.open(OrientDBEmbedded.java:48)
	at com.orientechnologies.orient.core.db.OrientDB.open(OrientDB.java:209)
	at com.orientechnologies.orient.core.db.OrientDB.open(OrientDB.java:194)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraphFactory.getDatabase(OrientGraphFactory.java:153)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraph.<init>(OrientGraph.java:145)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraph.<init>(OrientGraph.java:140)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraphFactory.getGraph(OrientGraphFactory.java:96)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraphFactory.getTx(OrientGraphFactory.java:85)
	at org.apache.tinkerpop.gremlin.orientdb.OrientGraphFactory.getTx(OrientGraphFactory.java:89)
	at com.callidus.mlm.config.MLMOrientDbConfig.getOrientGraph(MLMOrientDbConfig.java:90)
	at com.callidus.mlm.config.MLMOrientDbConfig$$EnhancerBySpringCGLIB$$8776ee0e.CGLIB$getOrientGraph$1(<generated>)
	at com.callidus.mlm.config.MLMOrientDbConfig$$EnhancerBySpringCGLIB$$8776ee0e$$FastClassBySpringCGLIB$$4bb2be3b.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:358)
	at com.callidus.mlm.config.MLMOrientDbConfig$$EnhancerBySpringCGLIB$$8776ee0e.getOrientGraph(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1173)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1067)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:325)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.obtainBeanInstanceFromFactory(ConfigurationClassEnhancer.java:389)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:361)
	at com.callidus.mlm.config.MLMOrientDbConfig$$EnhancerBySpringCGLIB$$8776ee0e.getOrientGraph(<generated>)
	at com.callidus.mlm.data.dao.impl.BaseDaoImpl.getOrientGraph(BaseDaoImpl.java:44)
	at com.callidus.mlm.service.internal.CompanyServiceImpl.createUpdate(CompanyServiceImpl.java:218)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:333)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:85)
	at com.callidus.mlm.audit.ApplicationAuditor.serviceMethodExecutionTime(ApplicationAuditor.java:42)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:629)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:618)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:213)
	at com.sun.proxy.$Proxy87.createUpdate(Unknown Source)
	at com.callidus.mlm.resource.CompanyResource.createUpdate(CompanyResource.java:143)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:133)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:116)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:230)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:474)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:349)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:783)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:798)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1434)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	- locked <0x000000078d11fe58> (a org.apache.tomcat.util.net.NioEndpoint$NioSocketWrapper)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- <0x000000078d100ce0> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"DestroyJavaVM" #49 prio=5 os_prio=0 tid=0x000000001cf97000 nid=0x3b20 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"http-nio-8084-AsyncTimeout" #47 daemon prio=5 os_prio=0 tid=0x000000001cf96800 nid=0x3cd0 waiting on condition [0x000000001e13f000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at java.lang.Thread.sleep(Native Method)
	at org.apache.coyote.AbstractProtocol$AsyncTimeout.run(AbstractProtocol.java:1138)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"http-nio-8084-Acceptor-0" #46 daemon prio=5 os_prio=0 tid=0x000000001cf95800 nid=0x18e4 runnable [0x000000002e68f000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.ServerSocketChannelImpl.accept0(Native Method)
	at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:422)
	at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:250)
	- locked <0x000000078d2591a0> (a java.lang.Object)
	at org.apache.tomcat.util.net.NioEndpoint$Acceptor.run(NioEndpoint.java:443)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"http-nio-8084-ClientPoller-1" #45 daemon prio=5 os_prio=0 tid=0x000000001cf95000 nid=0x3730 runnable [0x000000002e04e000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll0(Native Method)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll(WindowsSelectorImpl.java:296)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.access$400(WindowsSelectorImpl.java:278)
	at sun.nio.ch.WindowsSelectorImpl.doSelect(WindowsSelectorImpl.java:159)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x000000078d1bc260> (a sun.nio.ch.Util$3)
	- locked <0x000000078d1bc250> (a java.util.Collections$UnmodifiableSet)
	- locked <0x000000078d1bc010> (a sun.nio.ch.WindowsSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at org.apache.tomcat.util.net.NioEndpoint$Poller.run(NioEndpoint.java:775)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"http-nio-8084-ClientPoller-0" #44 daemon prio=5 os_prio=0 tid=0x000000001cf94000 nid=0x3bc0 runnable [0x000000002d4ee000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll0(Native Method)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll(WindowsSelectorImpl.java:296)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.access$400(WindowsSelectorImpl.java:278)
	at sun.nio.ch.WindowsSelectorImpl.doSelect(WindowsSelectorImpl.java:159)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x000000078d259a18> (a sun.nio.ch.Util$3)
	- locked <0x000000078d259a08> (a java.util.Collections$UnmodifiableSet)
	- locked <0x000000078d259898> (a sun.nio.ch.WindowsSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at org.apache.tomcat.util.net.NioEndpoint$Poller.run(NioEndpoint.java:775)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"NioBlockingSelector.BlockPoller-1" #43 daemon prio=5 os_prio=0 tid=0x000000001cf93800 nid=0xae8 runnable [0x000000002df0f000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll0(Native Method)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.poll(WindowsSelectorImpl.java:296)
	at sun.nio.ch.WindowsSelectorImpl$SubSelector.access$400(WindowsSelectorImpl.java:278)
	at sun.nio.ch.WindowsSelectorImpl.doSelect(WindowsSelectorImpl.java:159)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
	- locked <0x000000078d2550f0> (a sun.nio.ch.Util$3)
	- locked <0x000000078d2550e0> (a java.util.Collections$UnmodifiableSet)
	- locked <0x000000078d254f80> (a sun.nio.ch.WindowsSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
	at org.apache.tomcat.util.net.NioBlockingSelector$BlockPoller.run(NioBlockingSelector.java:339)

   Locked ownable synchronizers:
	- None

"Live Reload Server" #42 daemon prio=5 os_prio=0 tid=0x000000001cf92800 nid=0x21e0 runnable [0x000000002dbde000]
   java.lang.Thread.State: RUNNABLE
	at java.net.DualStackPlainSocketImpl.accept0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketAccept(DualStackPlainSocketImpl.java:131)
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:409)
	at java.net.PlainSocketImpl.accept(PlainSocketImpl.java:199)
	- locked <0x00000006c5a91f20> (a java.net.SocksSocketImpl)
	at java.net.ServerSocket.implAccept(ServerSocket.java:545)
	at java.net.ServerSocket.accept(ServerSocket.java:513)
	at org.springframework.boot.devtools.livereload.LiveReloadServer.acceptConnections(LiveReloadServer.java:157)
	at org.springframework.boot.devtools.livereload.LiveReloadServer.access$100(LiveReloadServer.java:45)
	at org.springframework.boot.devtools.livereload.LiveReloadServer$2.run(LiveReloadServer.java:126)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"File Watcher" #40 daemon prio=5 os_prio=0 tid=0x000000001cf92000 nid=0x3ad8 waiting on condition [0x000000002e46f000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at java.lang.Thread.sleep(Native Method)
	at org.springframework.boot.devtools.filewatch.FileSystemWatcher$Watcher.scan(FileSystemWatcher.java:255)
	at org.springframework.boot.devtools.filewatch.FileSystemWatcher$Watcher.run(FileSystemWatcher.java:239)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"OrientDB ONetworkProtocolHttpDb listen at 0.0.0.0:2480-2490" #38 prio=5 os_prio=0 tid=0x000000001cf91000 nid=0x3850 runnable [0x000000002da9e000]
   java.lang.Thread.State: RUNNABLE
	at java.net.DualStackPlainSocketImpl.accept0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketAccept(DualStackPlainSocketImpl.java:131)
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:409)
	at java.net.PlainSocketImpl.accept(PlainSocketImpl.java:199)
	- locked <0x00000006c596e2e8> (a java.net.SocksSocketImpl)
	at java.net.ServerSocket.implAccept(ServerSocket.java:545)
	at java.net.ServerSocket.accept(ServerSocket.java:513)
	at com.orientechnologies.orient.server.network.OServerNetworkListener.run(OServerNetworkListener.java:188)

   Locked ownable synchronizers:
	- None

"OrientDB ONetworkProtocolBinary listen at 0.0.0.0:2424-2430" #36 prio=5 os_prio=0 tid=0x000000001cf8c800 nid=0x3180 runnable [0x000000002abcf000]
   java.lang.Thread.State: RUNNABLE
	at java.net.DualStackPlainSocketImpl.accept0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketAccept(DualStackPlainSocketImpl.java:131)
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:409)
	at java.net.PlainSocketImpl.accept(PlainSocketImpl.java:199)
	- locked <0x00000006c59c5380> (a java.net.SocksSocketImpl)
	at java.net.ServerSocket.implAccept(ServerSocket.java:545)
	at java.net.ServerSocket.accept(ServerSocket.java:513)
	at com.orientechnologies.orient.server.network.OServerNetworkListener.run(OServerNetworkListener.java:188)

   Locked ownable synchronizers:
	- None

"Timer-0" #32 daemon prio=5 os_prio=0 tid=0x000000001cf8f800 nid=0x3d70 in Object.wait() [0x000000002d62f000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000006c59d46f0> (a java.util.TaskQueue)
	at java.util.TimerThread.mainLoop(Timer.java:552)
	- locked <0x00000006c59d46f0> (a java.util.TaskQueue)
	at java.util.TimerThread.run(Timer.java:505)

   Locked ownable synchronizers:
	- None

"MLMActorSystem-akka.actor.default-dispatcher-4" #29 prio=5 os_prio=0 tid=0x000000002559c000 nid=0x1d28 waiting on condition [0x00000000283df000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000006c5aac6e0> (a akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinPool)
	at scala.concurrent.forkjoin.ForkJoinPool.scan(ForkJoinPool.java:2075)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

   Locked ownable synchronizers:
	- None

"MLMActorSystem-scheduler-1" #26 prio=5 os_prio=0 tid=0x000000001f345000 nid=0x1c5c waiting on condition [0x00000000245be000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at java.lang.Thread.sleep(Native Method)
	at akka.actor.LightArrayRevolverScheduler.waitNanos(LightArrayRevolverScheduler.scala:85)
	at akka.actor.LightArrayRevolverScheduler$$anon$4.nextTick(LightArrayRevolverScheduler.scala:266)
	at akka.actor.LightArrayRevolverScheduler$$anon$4.run(LightArrayRevolverScheduler.scala:236)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"container-0" #25 prio=5 os_prio=0 tid=0x0000000023a07800 nid=0x3a78 waiting on condition [0x00000000246be000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at java.lang.Thread.sleep(Native Method)
	at org.apache.catalina.core.StandardServer.await(StandardServer.java:427)
	at org.springframework.boot.context.embedded.tomcat.TomcatEmbeddedServletContainer$1.run(TomcatEmbeddedServletContainer.java:174)

   Locked ownable synchronizers:
	- None

"ContainerBackgroundProcessor[StandardEngine[Tomcat]]" #24 daemon prio=5 os_prio=0 tid=0x000000001ecc7000 nid=0x2494 waiting on condition [0x00000000244be000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at java.lang.Thread.sleep(Native Method)
	at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.run(ContainerBase.java:1355)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"Service Thread" #12 daemon prio=9 os_prio=0 tid=0x000000001d012000 nid=0x2630 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C1 CompilerThread2" #11 daemon prio=9 os_prio=2 tid=0x000000001cfa9000 nid=0x3bb8 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread1" #10 daemon prio=9 os_prio=2 tid=0x000000001cfa6800 nid=0xa64 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread0" #9 daemon prio=9 os_prio=2 tid=0x000000001cfa0000 nid=0x330c waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"JDWP Command Reader" #8 daemon prio=10 os_prio=0 tid=0x000000001cf80000 nid=0x3430 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"JDWP Event Helper Thread" #7 daemon prio=10 os_prio=0 tid=0x000000001cf7a000 nid=0x3dc4 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"JDWP Transport Listener: dt_socket" #6 daemon prio=10 os_prio=0 tid=0x000000001cf70000 nid=0x3840 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Attach Listener" #5 daemon prio=5 os_prio=2 tid=0x000000001cf64800 nid=0x1280 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Signal Dispatcher" #4 daemon prio=9 os_prio=2 tid=0x000000001bb6f800 nid=0x3294 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Finalizer" #3 daemon prio=8 os_prio=1 tid=0x000000001bb4e800 nid=0x3798 in Object.wait() [0x000000001cf5f000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000006c4a47280> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
	- locked <0x00000006c4a47280> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)

   Locked ownable synchronizers:
	- None

"Reference Handler" #2 daemon prio=10 os_prio=2 tid=0x000000001bb07800 nid=0x3394 in Object.wait() [0x000000001cbce000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x00000006c4a467a0> (a java.lang.ref.Reference$Lock)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
	- locked <0x00000006c4a467a0> (a java.lang.ref.Reference$Lock)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

   Locked ownable synchronizers:
	- None

"VM Thread" os_prio=2 tid=0x000000001bb00000 nid=0x2848 runnable 

"GC task thread#0 (ParallelGC)" os_prio=0 tid=0x0000000001d71800 nid=0x39b0 runnable 

"GC task thread#1 (ParallelGC)" os_prio=0 tid=0x0000000001d73000 nid=0x2820 runnable 

"GC task thread#2 (ParallelGC)" os_prio=0 tid=0x0000000001d74800 nid=0x334 runnable 

"GC task thread#3 (ParallelGC)" os_prio=0 tid=0x0000000001d76000 nid=0x3e58 runnable 

"VM Periodic Task Thread" os_prio=2 tid=0x000000001d079800 nid=0x3428 waiting on condition 

JNI global references: 11726

**Comment by laa** [comment]
Guys, please do not put huge thread dumps inside of issue itself, because after that it is quite inconvenient to track history of such issues. We always can embed big amount of texts inside gist service.

**Comment by careerscale** [comment]
I think this was older snapshot. It seems good with snapshot no. 708.
